test/threads_exec.jl

```diff
@@ -672,3 +672,16 @@ let ch = Channel{Char}(0), t
     schedule(t)
     @test String(collect(ch)) == "hello"
 end
+
+@testset "Issue #32873" begin
+    m = Mutex()
+    try
+        lock(m)
+        lock(m)
+    catch ex
+        @test ex isa ErrorException
```

Comment:
Is it that this is not meant to be user-facing? and that's why it just a plain old `ErrorException`?

Reply:
(not that it'd be up to this PR to change anything)

Reply:
No idea, the error is just emitted using `error`, which means it's an `ErrorException`. :man_shrugging:

Reply:
ok 🤷‍♂